tests/expose_test.go

```diff
@@ -435,29 +796,48 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 				By("Repeating the sequence as prior to restarting the VM: Connect to exposed ClusterIP service.")
 				By("Starting a job which tries to reach the VMI via ClusterIP.")
 				job = runHelloWorldJob(serviceIP, servicePort, vmObj.Namespace)
+				jobCleanupFuncs = append(jobCleanupFuncs, cleanupJob(job.GetName(), job.GetNamespace()))
 
 				By("Waiting for the job to report a successful connection attempt.")
 				Expect(tests.WaitForJobToSucceed(job, 120*time.Second)).To(Succeed())
-			})
+			},
+				table.Entry("over default IPv4 IP family", k8sv1.IPv4Protocol),
+				table.Entry("over IPv6 IP family", k8sv1.IPv6Protocol),
+			)
+
+			table.DescribeTable("[test_id:343][label:masquerade_binding_connectivity]Should Verify an exposed service of a VM is not functional after VM deletion.", func(ipFamily k8sv1.IPFamily) {
+				if ipFamily == k8sv1.IPv6Protocol {
```

Comment:
lots of common code with line 744 ?
can we create some helper functions ?

Reply:
yep, there is a block that appear around 3 times, just to find the biggest common snippet, add parameters if needed, and reduce code repetitions please

Reply:
Done.

Reply:
I take you mean 

```
				By("Calling the start command on the stopped VM")
				vmi := startVMIFromVMTemplate(vm.GetName(), vm.GetNamespace())
				Expect(vmi).NotTo(BeNil(), "should have been able to create a VMI from a VM")
				tests.GenerateHelloWorldServer(vmi, testPort, "tcp")
```

Those are 4 lines of code, out of which 1 is a `by` clause, and another an expectation. 

That leaves 2 lines of code, one that starts a VM, another that creates a server within it - with error handling in the middle. Encapsulating that in another function, imho, would be counter-productive since the function does not make much sense; what would you name it ? `startVMAndCreateAServerInIt` ? That 'and' in the middle is a clear indication it should be 2 functions, not one. 

Would you say the same if the code was

```golang
tests.GenerateHelloWorldServer(
    startVMIFromVMTemplate(vm.GetName(), vm.GetNamespace()),
    testPort,
    "tcp")
```
instead ?

Reply:
> I take you mean
> 
> ```
> 				By("Calling the start command on the stopped VM")
> 				vmi := startVMIFromVMTemplate(vm.GetName(), vm.GetNamespace())
> 				Expect(vmi).NotTo(BeNil(), "should have been able to create a VMI from a VM")
> 				tests.GenerateHelloWorldServer(vmi, testPort, "tcp")
> ```
> 
> Those are 4 lines of code, out of which 1 is a `by` clause, and another an expectation.
> 
> That leaves 2 lines of code, one that starts a VM, another that creates a server within it - with error handling in the middle. Encapsulating that in another function, imho, would be counter-productive since the function does not make much sense; what would you name it ? `startVMAndCreateAServerInIt` ? That 'and' in the middle is a clear indication it should be 2 functions, not one.

`startVMWithServer`
but up to you if to do it or leave it as is.

> 
> Would you say the same if the code was
> 
> ```go
> tests.GenerateHelloWorldServer(
>     startVMIFromVMTemplate(vm.GetName(), vm.GetNamespace()),
>     testPort,
>     "tcp")
> ```
> 
> instead ?

nope, this one is self contained and reuse code.

Reply:
> `startVMWithServer`

Ah... Now it was not so hard ... 0_o 

I'll keep it as is since it allows me to do error handling in the test case body, where it belongs.